src/charm.py

```diff
@@ -391,6 +391,11 @@ def _configure(self):  # noqa: C901
             self._stored.status["config"] = to_tuple(MaintenanceStatus("Configuring Loki"))
             return
 
+        if 0 < int(self.config["retention-period"]) < 24:
```

Comment:
is this an upstream limitation? Could we use a duration string (i.e. `2d 4h`, `2M`, '4w' etc.) here instead to improve the UX for the operator?

Reply:
So loki allows setting duration strings with the exception of `M`, but my initial thinking to this was to guard from any input errors for example people putting `1h` or `3600s` cause Loki itself doesn't really panic if value is less than `24h` and automatically defaults to `744h` and putting duration string validation in the charm itself didn't really seem like the best thing to do.

the upstream uses [Prometheus type duration](https://github.com/prometheus/common/blob/e54e4df4b9182e2fceac62aa447b507a53e9114d/model/time.go#L189)

```
var unitMap = map[string]struct {
	pos  int
	mult uint64
}{
	"ms": {7, uint64(time.Millisecond)},
	"s":  {6, uint64(time.Second)},
	"m":  {5, uint64(time.Minute)},
	"h":  {4, uint64(time.Hour)},
	"d":  {3, uint64(24 * time.Hour)},
	"w":  {2, uint64(7 * 24 * time.Hour)},
	"y":  {1, uint64(365 * 24 * time.Hour)},
}
```

Reply:
this however can be avoided and changed to string without putting the validation in the charm if we maybe just add it in the config docstrings that if any value is put less than `24h` workload won't panic yet won't honor it and will default to `744h`

Reply:
Also when I say validation inside the charm I mean smth like below

```
def duration_to_hours(duration_str):
    total_hours = 0
    matches = re.findall(r'(\d+)([a-zA-Z]+)', duration_str)
    conversion_factors = {
        "ms": 1 / (1000 * 3600), 
        "s": 1 / 3600,            
        "m": 1 / 60,               
        "h": 1,                   
        "d": 24,                  
        "w": 24 * 7, 
        "y": 24 * 365             
    }
    supported_types = ", ".join(conversion_factors.keys())
    for match in matches:
        print(match)
        value = float(match[0])
        unit = match[1].lower()
        if unit in conversion_factors:
            total_hours += value * conversion_factors[unit]
        else:
            return f"Unsupported duration type: {unit}. Supported types are: {supported_types}"
    return int(total_hours)

retention_period = "1y1w1d4h61m"
```

Reply:
An int can also be negative. To we need the `0 <`?

Reply:
Kept as an `int` config yet switched the workload to switch it to days instead of hours